tsnet/tsnet.go

```diff
@@ -944,6 +946,32 @@ func (s *Server) APIClient() (*tailscale.Client, error) {
 	return c, nil
 }
 
+// AuthenticatedAPITransport provides an HTTP transport that can be used with
+// the control server API without needing additional authentication details. It
+// authenticates using the current client's nodekey.
+//
+// For example:
+//
+//	import "net/http"
+//	import "tailscale.com/client/tailscale/v2"
+//	import "tailscale.com/tsnet"
+//
+//	var s *tsnet.Server
+//	...
+//	rt, err := s.AuthenticatedAPITransport()
+//	// handler err ...
+//	var client tailscale.Client{HTTP: http.Client{
+//	    Timeout: 1*time.Minute,
+//	    UserAgent: "your-useragent-here",
+//	    Transport: rt,
+//	}}
+func (s *Server) AuthenticatedAPITransport() (http.RoundTripper, error) {
```

Comment:
I think to unstick this PR, it's probably best to initially either delete this or require `I_Acknowledge_This_API_Is_Unstable` to call it. We can easily wind that decision back in a future PR, but it's harder to go the other way.

Reply:
I created `I_Acknowledge_This_API_Is_Experimental` inside this package to avoid forcing a dependency on the now deprecated `client/tailscale` package.